app/services/users/login_service.rb

```diff
@@ -61,18 +63,25 @@ def set_autologin_cookie
       return unless Setting::Autologin.enabled?
 
       # generate a key and set cookie if autologin
-      expires_on =  Setting.autologin.days.from_now.beginning_of_day
-      token = Token::AutoLogin.create(user:, data: session_identification, expires_on:)
+      token = Token::AutoLogin.create(user:, data: token_session_information)
```

Comment:
:yellow_circle: Not directly in the scope of this PR, but would you agree that we do:

* not expect the creation of the token to fail (specifically due to validation errors)
* not check whether the token was persisted

If so, I'd suggest that we replace `create` with `create!` here, which will raise an error if a validation error would occur.

```suggestion
      token = Token::AutoLogin.create!(user:, data: token_session_information)
```

Reply:
I actually had the bang-style first for the session link, only to then notice that we didn't do it in this line too. Happy to change both the other way around